src/lib/mongoBackend/mongoUpdateContextSubscription.cpp

```diff
@@ -208,34 +207,43 @@ HttpStatusCode mongoUpdateContextSubscription
   }
   responseP->subscribeResponse.subscriptionId = requestP->subscriptionId;
 
-  reqSemGive(__FUNCTION__, "ngsi10 update subscription request", reqSemTaken);
 
 
   //
   // Modification of the subscription cache
   //
-  // The subscription before this updatye is stored in 'sub'.
-  // The updated subscription is stored in 'newSub'
+  // The subscription "before this update" is looked up in cache and referenced by 'cSubP'.
+  // The "updated subscription information" is in 'newSubObject' (mongo BSON object format).
   // 
   // All we need to do now for the cache is to:
-  //   1. Remove 'sub' from sub-cache (if present)
-  //   2. Create 'newSub' in sub-cache (if applicable)
+  //   1. Remove 'cSubP' from sub-cache (if present)
+  //   2. Create 'newSubObject' in sub-cache (if applicable)
+  //
+  // The subscription is already updated in mongo.
+  //
+  // FIXME P7: mongoSubCache stuff should be avoided if subscription is not patterned
   //
 
   // 0. Lookup matching subscription in subscription-cache
-  std::string servicePath;
+  CachedSubscription* cSubP = mongoSubCacheItemLookup(tenant.c_str(), requestP->subscriptionId.get().c_str());
 
-  servicePath = (servicePathV.size() == 0)? "" : servicePathV[0];
-  orion::Subscription* subP = orion::subCache->lookupById(tenant, servicePath, requestP->subscriptionId.get());
-
-  // 1. Remove 'sub' from sub-cache (if present)
-  if (subP)
+  if (cSubP == NULL)
```

Comment:
As discussed as skype, this need a fix to cover cases in which the original subscription hasn't pattern (so cSupP is NULL) but the new subscriptions has patterns.